src/parquet.cc

```diff
@@ -33,8 +33,20 @@ using parquet::PageType;
 using parquet::SchemaElement;
 using parquet::Type;
 
-InMemoryInputStream::InMemoryInputStream(const uint8_t* buffer, int64_t len) :
-  buffer_(buffer), len_(len), offset_(0) {
+InMemoryInputStream::InMemoryInputStream(int64_t len) : len_(len), offset_(0) {
+  buffer_ = (uint8_t*)malloc(sizeof(uint8_t) * len_);
+}
+
+InMemoryInputStream::~InMemoryInputStream() {
+  free(buffer_);
+}
```

Comment:
IMHO, it's dangerous to manage the life cycle of a memory buffer outside of the object that's using it. Anyway, instead of modifying InMemoryInputStream, I can add another InputStream that manages its own memory.